File: 190-ReverseBits/190-ReverseBits.cpp
// Last updated: 6/11/2026, 11:35:17 AM
class Solution {
public:
    int reverseBits(int n) {
        int tot = 31;
        int ans = 0;
        for(int i=0; i<31; i++){
            if((n&(1LL<<i))){
                ans |= (1LL<<(tot-i));
            }
        }
        return ans;
    }
};

File: 190-ReverseBits/190-ReverseBits.py
# Last updated: 6/11/2026, 11:35:21 AM
class Solution:
    def reverseBits(self, n: int) -> int:
        multiplier = 2**31
        tot_sum = 0
        while(n):
            last_digit = n&1
            tot_sum += multiplier*last_digit
            n = n >> 1
            multiplier = multiplier//2
            print(tot_sum)
        return tot_sum